subscriptions-api.md

```diff
@@ -0,0 +1,613 @@
+other settings properties SHOULD be supported by all implementations.
+
+- **address** (string) – OPTIONAL. The link target node in the
+  AMQP container identified by the sink URI, if not expressed in the sink URI's
+  path portion.
+- **linkname** (string) – OPTIONAL. Name to use for the AMQP link. If not set,
+  a random link name is used.
+- **sendersettlementmode** (string) – OPTIONAL. Allows to control the sender's
+  settlement mode, which determines whether transfers are performed "settled"
+  (without acknowledgement) or "unsettled" (with acknowledgement). Default
+  value is unsettled.
+- **linkproperties** (map) – OPTIONAL. A set of key/value pairs that are
+  copied into link properties for the send link.
+
+##### 3.2.2.4. Apache Kafka
+
+All implementations that support Apache Kafka MUST support the _topicname_
+settings. All other settings SHOULD be supported.
+
+- **topicname** (string) - REQUIRED. The name of the Kafka topic to publish to.
+- **partitionkeyextractor** (string) - OPTIONAL. A partition key extractor
+  expression per the CloudEvents Kafka transport binding specification.
+- **clientid** (string)
+- **acks** (string)
+
+##### 3.2.2.5. NATS
+
+- **subject** (string) - REQUIRED. The name of the NATS subject to publish to.
+
+#### 3.2.3 Filter Dialects
+
+The filter expression language supported by an event producer is indicated by
+its dialect. This is intended to allow for flexibility, extensibility and to
+allow for a variety of filter dialects without enumerating them all in this spec or
+predicting what filtering needs every system will have in the future. This spec
+will specify a single "simple" dialect, which all implementations MUST support.
+
+The dialect for a particular filter is indicated by specifying the "dialect"
+property at the root of the JSON object. The value of this is a string encoded
+unique identifier for the filter dialect. Subscriptions specifying the "filter"
+property MUST specify a dialect. All other properties are dependent on the
+dialect being used.
+
+##### 3.2.3.1. "simple" filter dialect
+
+The "simple" filter dialect is intended to support the most common filtering use
+cases:
+
+- "Exact match" where the filter condition and an attribute value match exactly.
+- "Prefix match" where the filter condition is a prefix of the attribute value.
+- "Suffix match" where the filter condition is a suffix of the attribute value.
+
+This filter dialect is intentionally constrained to these filter types, since
+filtering has a potentially significant impact on the baseline performance of
+all implementations. In cases where more filter types or different expression
+languages are desired, further dialects can be introduced as extensions.
+
+Extension dialects will have varying support across event producers. It is up to
+the subscriber and producer to negotiate which filter dialects can be used
+within a given subscription.
```

Comment:
When ready we might want to add a reference to the Discovery spec here since that should help with the negotiations.